mainchain-follower/db-sync-follower/src/candidates/mod.rs

```diff
@@ -313,14 +330,15 @@ impl CandidatesDataSourceImpl {
 				Some((IntegerDatum(p), IntegerDatum(t))) => {
 					p.to_u16().zip(t.to_u16()).map(|(p, t)| DParameter {
 						num_permissioned_candidates: p,
-						num_registered_candidates: t,
+						num_ada_candidates: t,
+						num_eth_candidates: 0,
```

Comment:
Could this decoder handle situations when DParam on Cardano has 3 values and 2 values? Only in the latter case it would use 0 as default?

Reply:
Yes, if we generalize a bit and declare it like this:
```rust
pub struct DParameter {
	pub num_permissioned_candidates: u16,
	pub num_ada_candidates: u16,
	pub num_other_candidates: Vec<u16>,  // one item for each chain
}

```
This is more universal encoding, so both empty vector and a vector with single element can be supported. It will also support for example adding PC native token to the virtual stake.

Reply:
just curious, will there be a new candidate selection algorithm?

Reply:
(also if this new structure remains we should consider renaming it)

Reply:
Yes, Minotaur "Mini", to replace Ariadne. Currently I assume that if num_other_candidates is empty, then it should behave as Ariadne. It may change in future, though.

Reply:
> (also if this new structure remains we should consider renaming it)

We need a good name, please suggest. 
People got used to DParameter and what it means, and the meaning will stay the same, maybe somewhat generalized.

Reply:
IMO (u16, u16, vec<u16>) doesn't help.
After all it could be a Vec only or fields only and there has to be connection between types of candidates and parts of D-Parameter. Please bare in mind that this part of code is reading data from Datum and this can be affected by how the data is written.

For the decision Vec vs Map vs named fields should be driven by shape of data and what are we building: a generalization to N types of candidates or generalization to 3 candidates types